neqo-http3/src/server.rs

```diff
@@ -938,11 +966,91 @@ mod tests {
     fn test_server_stop_sending_decoder_stream() {
         let (mut hconn, mut peer_conn) = connect();
         peer_conn
-            .conn
             .stream_stop_sending(SERVER_SIDE_DECODER_STREAM_ID, Error::HttpNoError.code())
             .unwrap();
-        let out = peer_conn.conn.process(None, now());
+        let out = peer_conn.process(None, now());
         hconn.process(out.dgram(), now());
         assert_closed(&mut hconn, &Error::HttpClosedCriticalStream);
     }
+
+    /// Perform a handshake, then another with the token from the first.
+    /// The second should always resume, but it might not always accept early data.
+    fn zero_rtt_with_settings(settings: QpackSettings, zero_rtt: &ZeroRttState) {
+        let (_, client) = connect();
+        let token = client.resumption_token();
+        assert!(token.is_some());
+
+        let mut server = create_server(settings);
+        let mut client = default_client();
+        client.set_resumption_token(now(), &token.unwrap()).unwrap();
+
+        connect_transport(&mut server, &mut client, true);
+        assert!(client.tls_info().unwrap().resumed());
+        assert_eq!(client.zero_rtt_state(), zero_rtt);
+    }
+
+    #[test]
+    fn zero_rtt() {
+        zero_rtt_with_settings(DEFAULT_SETTINGS, &ZeroRttState::AcceptedClient);
+    }
+
+    /// A larger QPACK decoder table size isn't an impediment to 0-RTT.
+    #[test]
+    fn zero_rtt_larger_decoder_table() {
+        zero_rtt_with_settings(
+            QpackSettings {
+                max_table_size_decoder: DEFAULT_SETTINGS.max_table_size_decoder + 1,
+                ..DEFAULT_SETTINGS
+            },
+            &ZeroRttState::AcceptedClient,
+        );
+    }
+
+    /// A smaller QPACK decoder table size prevents 0-RTT.
+    #[test]
+    fn zero_rtt_smaller_decoder_table() {
```

Comment:
@ddragana, not sure what you were asking for.  All the tests like this with Rejected are the ones where the 0-RTT check fails.

Reply:
sorry, my mistake.